src/point_one/fusion_engine/parsers/fusion_engine_framer.cc

```diff
@@ -70,12 +124,18 @@ void FusionEngineFramer::SetBuffer(void* buffer, size_t capacity_bytes) {
     capacity_bytes = 0x7FFFFFFF;
   }
 
+#if P1_HAVE_STD_SMART_PTR
   if (buffer == nullptr) {
     managed_buffer_.reset(new uint8_t[capacity_bytes]);
     buffer = managed_buffer_.get();
   } else if (buffer != managed_buffer_.get()) {
     managed_buffer_.reset(nullptr);
   }
+#else
+  if (buffer == nullptr) {
+    LOG(FATAL) << "P1_HAVE_STD_SMART_PTR required for dynamic memory.";
```

Comment:
In the no smart pointer case, we could just add an extra bool member var saying that we own the memory, and then `delete[]` the raw pointer in this class's destructor (and also the else if above). The smart pointer was just a convenience to get that same behavior. Honestly we could also just remove it in favor of the bool?

Reply:
The reason that smart pointers are needed is for all the implicit copy/move constructors. Getting the behavior correct without implicit RAII objects is always a pain. I wanted to avoid it since it adds a lot of additional code, but I guess a compromise would be to delete the implicit constructors.